tools/tests/test_doctrine_callout.py

```
@@ -654,3 +672,388 @@ def test_every_ceiling_the_callout_states_is_the_one_in_force(tmp_path, monkeypa
+    _write(tmp_path, {"alpha": "does a DIFFERENT thing"})
+    monkeypatch.setattr(dc, "ROOT", tmp_path)
+    gh(["AGENTS.md", f"{dc.roster.CELLS}/alpha/{dc.roster.CELL_FILE}"])
+    status, lines = dc.run("1", None, base="HEAD", dry_run=True)
+    assert status == dc.OK
+    touched = next(l for l in lines if "touches" in l)
+    # The path arm's member first, the frontmatter arm's appended.
+    assert "AGENTS.md" in touched
+    assert f"{dc.roster.CELLS}/alpha/{dc.roster.CELL_FILE}" in touched
+
+
+def test_the_charter_filter_is_read_at_both_of_its_use_sites(tmp_path):
```

Ruff (0.16.3)

[warning] Unused function argument: `tmp_path`

(ARG001)